### ISSUE TITLE
switch.eh123456_smart_charging does not reflect charger state correctly

### ISSUE THREAD

**Issue by uphillbattle** [fondberg/easee_hass#232]
### The problem

When toggling the HA `switch.eh123456_smart_charging` switch to `off` from the HA UI, the charger correctly disables smart charging (LEDs go white), but the HA `switch` reverts (after a second or two) to `on`. The charger state does not revert (LEDs still white).

If I toggle the HA `switch` to `off` once more, the charger stays in the smart charging disabled state (LEDs still white - nothing happens), and the HA `switch` also remains `off`.

If I then toggle the HA `switch` to `on`, the charger LEDs go blue and the HA `switch` stays `on`.

The problem only occurs, it seems, when toggling the HA `switch` to `off`.

![image](https://user-images.githubusercontent.com/15340540/196901412-265ffb7e-130a-4e77-a008-c90986d3a390.png)
![Screenshot_20221020-103929](https://user-images.githubusercontent.com/15340540/196902424-45a8d5e5-4ce8-49c3-8c1c-67dc4f6215c0.jpg)

### Version of Easee integration having the issue?

0.9.45

### Version of Home Assistant Core having the issue?

2022.10.4

**Comment by astrandb** [comment]
I can reproduce this with my charger. I will have a look.

**Comment by uphillbattle** [comment]
I haven't had the chance to check, but it will be interesting to check if disabling smart charging from the charger itself (pushing the button) is correctly shown by the HA `switch`. This is my actual use case: To be able to disable smart charging (controlled by HA) from the charger itself (if you want to override smart charging and just charge).

**Comment by astrandb** [comment]
It is the same behaviour with all config switches (`cable locked permanently`, `enable idle current` and `smart charging`) 
Turning off  and on from the UI  is handled correctly by the API and the charger, but turning off is not noted by the UI, repeating the turning off after a few seconds makes the UI switch stay in off position.
Extract from debug log:
```
2022-10-20 09:41:55.596 DEBUG (MainThread) [custom_components.easee.switch] smart_charging Switch turn off
2022-10-20 09:41:55.596 DEBUG (MainThread) [custom_components.easee.switch] Getting state of smart_charging
2022-10-20 09:41:55.597 DEBUG (MainThread) [pyeasee.easee] POST: /api/chargers/EH468215/settings ({'json': {'smartCharging': False}})
2022-10-20 09:41:55.597 DEBUG (MainThread) [pyeasee.easee] verify_updated_token: 2022-10-21 09:40:08.577140, 2022-10-20 09:41:55.597801, False
2022-10-20 09:41:57.364 DEBUG (MainThread) [custom_components.easee.controller] Callback EH468215 46 state_ledMode 23
2022-10-20 09:41:57.364 DEBUG (MainThread) [custom_components.easee.controller] Scheduling update
2022-10-20 09:41:57.380 DEBUG (MainThread) [custom_components.easee.controller] Callback EH468215 102 state_smartCharging True
2022-10-20 09:41:57.380 DEBUG (MainThread) [custom_components.easee.controller] Scheduling update
2022-10-20 09:41:57.396 DEBUG (MainThread) [custom_components.easee.controller] Callback EH468215 102 state_smartCharging True
2022-10-20 09:41:57.396 DEBUG (MainThread) [custom_components.easee.controller] Scheduling update
```

**Comment by uphillbattle** [comment]
Ah, good catch 👍 

So, an alternative way of checking if the charger is in fact in Smart or Normal mode is to read the attribute `state_ledMode` from `sensor.eh123456_status`. 
https://developer.easee.cloud/docs/enumerations

Is this also how the `switch` is set in the first place (when fetching from the API)?

**Comment by astrandb** [comment]
> s this also how the `switch` is set in the first place (when fetching from the API)?

No, correct value is reported directly after restart of the singnalr stream.

**Comment by olalid** [comment]
I can not test this where I am right now, but the general approach with these switches is that when we toggle them we change their state in the UI immediately and call the API to change the setting on the Easee servers.
So if that API call does for some reason fail it could be where the issue is.
The state is then later supposed to be confirmed by the servers with a message over signalr (which is what this callback message is doing).
If that callback message does not reflect the actual state I would say this is a problem on the server side, but if we for some reason are interpreting it incorrectly the problems is on our side.
Note that the server also sends out these messages periodically over signalr independant of it is changed or not, but it could be several hours between each time for unchanged observations.

**Comment by uphillbattle** [comment]
I notice that it takes a few seconds from I manipulate the `switch` in HA UI until the Easee app reports changed LED color. If the API reports the actual state of the charger, it will take a few seconds, I guess, from the `switch` manipulation until the API can report the correct state of the charger. If the charger state is reported immediately (over signalr), the API may still believe the charger is in the wrong state. Is it possible to schedule a state check a period of time (say, 10 seconds) after a `switch` manipulation to get the correct state?

**Comment by olalid** [comment]
The delay is probably the time it takes for the signalr message (or whatever they are using in the app, they have several different streaming interfaces) to be sent out and received.
In the log above we can see how smartcharing is set to false at 2022-10-20 09:41:55.597 and then a message is received about 2 seconds later 2022-10-20 09:41:57.380 that says it is true.
I would say this is quite clear evidence that the problem is on the server side.
In any case it is the changed state that is supposed to trigger that the new value is sent out over signalr, not the call to the API.
I think the correct thing to do here is to gather some "evidence" and tell Easee about it.
I can look at this earlist this evening, so if anyone else feel they want to go ahead.

**Comment by olalid** [comment]
Yes, this is definitely a problem on the server side. At startup signalr reports the correct value, but after the setting is changed it is always reported as True (i.e. turned on), even if it is turned off.
I will report this to Easee, I do not think we should make any workarounds at this point.

```
2022-10-20 21:19:37.428 DEBUG (MainThread) [custom_components.easee.switch] smart_charging Switch turn on
2022-10-20 21:19:37.428 DEBUG (MainThread) [custom_components.easee.switch] Getting state of smart_charging
2022-10-20 21:19:37.428 DEBUG (MainThread) [pyeasee.easee] POST: /api/chargers/EHxxxx/settings ({'json': {'smartCharging': True}})
2022-10-20 21:19:38.940 DEBUG (MainThread) [custom_components.easee.controller] Callback EHxxx 102 state_smartCharging True <class 'bool'>
2022-10-20 21:19:38.941 DEBUG (MainThread) [custom_components.easee.controller] Callback EHxxx 102 state_smartCharging True <class 'bool'>
2022-10-20 21:19:38.948 DEBUG (MainThread) [custom_components.easee.switch] Getting state of smart_charging
2022-10-20 21:19:38.951 DEBUG (MainThread) [custom_components.easee.switch] Getting state of smart_charging
2022-10-20 21:19:38.954 DEBUG (MainThread) [custom_components.easee.entity] Entity async_update : EHxxx smart_charging
2022-10-20 21:19:38.961 DEBUG (MainThread) [custom_components.easee.switch] Getting state of smart_charging
2022-10-20 21:19:56.711 DEBUG (MainThread) [custom_components.easee.switch] smart_charging Switch turn off
2022-10-20 21:19:56.711 DEBUG (MainThread) [custom_components.easee.switch] Getting state of smart_charging
2022-10-20 21:19:56.711 DEBUG (MainThread) [pyeasee.easee] POST: /api/chargers/EHxxx/settings ({'json': {'smartCharging': False}})
2022-10-20 21:19:57.935 DEBUG (MainThread) [custom_components.easee.controller] Callback EHxxx 102 state_smartCharging True <class 'bool'>
2022-10-20 21:19:57.935 DEBUG (MainThread) [custom_components.easee.controller] Callback EHxxx 102 state_smartCharging True <class 'bool'>
2022-10-20 21:19:57.940 DEBUG (MainThread) [custom_components.easee.switch] Getting state of smart_charging
2022-10-20 21:19:57.945 DEBUG (MainThread) [custom_components.easee.switch] Getting state of smart_charging
2022-10-20 21:19:57.946 DEBUG (MainThread) [custom_components.easee.entity] Entity async_update : EHxxx smart_charging
```

**Comment by uphillbattle** [comment]
Good job! 👍 Thank you for reporting to Easee!

**Comment by olalid** [comment]
I did a small test program to try and make it easy to replicate the problem for Easee since they could not, but I then discovered that the problem seems to be that they have changed how boolean values are transferred, sometimes...
When the signalr first connects and the startup values are transferred, the boolean values are transferred as 0, which we interpret as False.
Then, later when we change the state, they transfer the state as "True" or "False". We interpret both of these as True, since we are expecting "0" or "1" and our logic is to interpret anything else than "0" as True.
So we can fix this, but we need to make a new version of the lib...
But I will first ask them why they do this like this :)
